### PR TITLE
Add MetricProfile In-Memory Storage Collection

### PR DESCRIPTION
**Pull request by shreyabiradar07**
## Description

This PR implements in-memory storage of `MetricProfile` during Autotune initialization and after creating a new metric profile

NOTE: this PR is created on top of #1258 and has a dependency to be merged first

### Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Docs update
- [ ] Breaking change (What changes might users need to make in their application due to this PR?)
- [ ] Requires DB changes

## How has this been tested?

Tested using local_monitoring demo script

**Test Configuration**
* Kubernetes clusters tested on: minikube, openshift

## Checklist :dart:

- [x] Followed coding guidelines
- [x] Comments added
- [ ] Dependent changes merged
- [ ] Documentation updated
- [ ] Tests added or updated

## Additional information
.

### CODE REVIEW

File: src/main/java/com/autotune/analyzer/performanceProfiles/MetricProfileCollection.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        LOGGER.info(KruizeConstants.MetricProfileConstants.ADDING_METRIC_PROFILE + "{}", metricProfileName);
+
+        if(metricProfileCollection.containsKey(metricProfileName)) {
+            LOGGER.error(KruizeConstants.MetricProfileConstants.METRIC_PROFILE_ALREADY_EXISTS + "{}", metricProfileName);
+        } else {
+            LOGGER.info(KruizeConstants.MetricProfileConstants.METRIC_PROFILE_ADDED + "{}", metricProfileName);
+            metricProfileCollection.put(metricProfileName, metricProfile);
+        }
+    }
+}
```
Comment (msvinaykumar) [inline src/main/java/com/autotune/analyzer/performanceProfiles/MetricProfileCollection.java:75]: eof missing

File: src/main/java/com/autotune/database/table/KruizeMetricProfileEntry.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    public JsonNode getSlo() {
+        return slo;
+    }
+
+    public void setSlo(JsonNode slo) {
+        this.slo = slo;
+    }
+}
```
Comment (msvinaykumar) [inline src/main/java/com/autotune/database/table/KruizeMetricProfileEntry.java:107]: eof missing

File: src/main/java/com/autotune/analyzer/services/MetricProfileService.java [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+    public void sendErrorResponse(HttpServletResponse response, Exception e, int httpStatusCode, String errorMsg) throws
+            IOException {
+        if (null != e) {
+            LOGGER.error(e.toString());
+            if (null == errorMsg)
+                errorMsg = e.getMessage();
+        }
+        response.sendError(httpStatusCode, errorMsg);
+    }
+}
```
Comment (msvinaykumar) [inline src/main/java/com/autotune/analyzer/services/MetricProfileService.java:248]: eof is missing

**Review by msvinaykumar — APPROVED** [review]
Review Checklist

#### Common Review
- [ ] Code formatting adheres to the project’s style guide.
- [x] No unnecessary commented-out code.
- [ ] Appropriate use of comments and documentation.

#### Logical Flow Review
- [ ] Code logic is clear and easy to follow.
- [ ] All branches and conditions are covered and tested.
- [ ] No redundant or duplicate code.

#### Demo Update
- [ ] Demonstration examples are updated to reflect changes.
- [ ] README and documentation include updated demo instructions.
- [ ] Any necessary demo files are included and tested.

#### Naming Conventions
- [ ] Variable and function names follow the project's naming conventions.
- [ ] File names are descriptive and adhere to naming standards.
- [ ] No ambiguous or misleading names.

#### Workflow
- [ ] CI/CD pipelines are updated and passing.
- [ ] Tests are included and all pass.
- [ ] The workflow for deploying or integrating changes is clear and documented.

#### Does It Affect ROS 
- [ ] Changes are reviewed for compatibility with ROS.
- [ ] ROS-specific configurations are updated if necessary.
- [ ] Any ROS dependencies or integrations are tested and verified.

**Comment by msvinaykumar** [comment]
LGTM

**Review by chandrams — APPROVED** [review]
LGTM

File: src/main/java/com/autotune/analyzer/services/MetricProfileService.java (hunk 3 above)
Comment (shreyabiradar07) [reply src/main/java/com/autotune/analyzer/services/MetricProfileService.java]: done

**Review by msvinaykumar — APPROVED** [review]
Review Checklist

#### Common Review
- [x] Code formatting adheres to the project’s style guide.
- [x] No unnecessary commented-out code.
- [x] Appropriate use of comments and documentation.

#### Logical Flow Review
- [x] Code logic is clear and easy to follow.
- [x] All branches and conditions are covered and tested.
- [x] No redundant or duplicate code.

#### Demo Update
- [x] Demonstration examples are updated to reflect changes.
- [x] README and documentation include updated demo instructions.
- [x] Any necessary demo files are included and tested.

#### Naming Conventions
- [x] Variable and function names follow the project's naming conventions.
- [x] File names are descriptive and adhere to naming standards.
- [x] No ambiguous or misleading names.

#### Workflow
- [ ] CI/CD pipelines are updated and passing.
- [ ] Tests are included and all pass.
- [x] The workflow for deploying or integrating changes is clear and documented.

#### Does It Affect ROS 
- [ ] Changes are reviewed for compatibility with ROS.
- [x] ROS-specific configurations are updated if necessary.
- [ ] Any ROS dependencies or integrations are tested and verified.

**Comment by msvinaykumar** [comment]
LGTM

**Pull request merged**